src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -389,6 +451,74 @@ private static string GetEscapedParameterName(AttributeData attribute, string pa
         }
     }
 
+    private static bool TryGetAsParametersConstructor(INamedTypeSymbol type, out bool? isParameterlessConstructor, [NotNullWhen(true)] out IEnumerable<ConstructorParameter>? matchedParameters)
```

Comment:
Without having thought deeply about it, it's not obvious to me why you'd need a three-state to indicate whether or not `matchedParameters` is empty.

Reply:
Also, wouldn't it be more useful to return a collection type with a length?

Reply:
> Without having thought deeply about it, it's not obvious to me why you'd need a three-state to indicate whether or not matchedParameters is empty.

I think [the comment here](https://github.com/dotnet/aspnetcore/pull/47914#discussion_r1187033285) provides some additional insight but these are the scenarios that `TryGetAsParameters` processes:

- Returns false when an invalid constructor or no constructor is found. This will result in error diagnostics being emitted at build.
- Returns true with `isParameterlessConstructor`/`useDefaultConstructor` set to `true` which indicates that the default parameterless constructor with the values contained in `matchedParameters`/`matchedProperties` being passed to the object initializer should be used.
- Returns true with `isParameterlessConstructor`/`useDefaultConstructor` set to `fase` which indicates that a parameterized constructor with the values contained in `matchedParameters`/`matchedProperties` being passed to it should be used.

> Also, wouldn't it be more useful to return a collection type with a length?

Not really? We don't really need to know the length here since we're enumerated the matched values with the purpose of producing a string for codegen.

Reply:
I think I remember the caller checking the length, but I might be thinking of something else.